comRamona/assignment1/Question1.java

```diff
@@ -0,0 +1,132 @@
+        }
+        if (firstString.length() != secondString.length()) {
+            return false;
+        }
+
+        if (!isCaseSensitive) {
+            firstString = firstString.toLowerCase();
+            secondString = secondString.toLowerCase();
+        }
+
+        if (firstString.equals(secondString)) {
+            return true;
+        }
+
+        // use a hashMap for the frequencies of characters in the first word, compare
+        // to the characters in the second word
+        Map<Character, Integer> map_firstWord = new HashMap<>();
+        for (int i = 0; i < firstString.length(); i++) {
+            char c = firstString.charAt(i);
+            map_firstWord.put(c, map_firstWord.getOrDefault(c, 0) + 1);
+        }
+
+        for (int i = 0; i < secondString.length(); i++) {
+            char c = secondString.charAt(i);
+            int freq = map_firstWord.getOrDefault(c, 0);
+            if (freq <= 0) {
+                return false;
+            }
+            map_firstWord.put(c, freq - 1);
+        }
+
+        return true;
+    }
+
+    /**
+     * @param firstSentence
+     * @param secondSentence
+     * @param isCaseSensitive
+     * @return true if every word in the first sentence has an anagram in the second sentence
+     */
+    public static boolean areSentencesAnagrams(String firstSentence, String secondSentence, boolean isCaseSensitive) {
+        if (firstSentence == null || secondSentence == null) {
+            return false;
+        }
+        if (!isCaseSensitive) {
+            firstSentence = firstSentence.toLowerCase();
+            secondSentence = secondSentence.toLowerCase();
+        }
+
+        String[] firstSentWords = firstSentence.split("\\W+");
+        String[] secondSentWords = secondSentence.split("\\W+");
+
+        if (firstSentWords.length != secondSentWords.length) {
+            return false;
+        }
+
+        Map<String, Integer> secondSentWordsMap = new HashMap<>();
+        for (String word : secondSentWords) {
+            secondSentWordsMap.put(word, secondSentWordsMap.getOrDefault(word, 0) + 1);
+        }
+
+        // check every word in the first sentence has an anagram in the second sentence
+        // use the hashmap of word frequencies of the second sentence for repeated words
+        for (String word : firstSentWords) {
+            boolean foundMatch = false;
+            for (String matchWord : secondSentWordsMap.keySet()) {
+                int freq = secondSentWordsMap.get(matchWord);
+                if (freq == 0) {
+                    continue;
+                }
+                if (areWordsAnagrams(word, matchWord, true)) {
+                    secondSentWordsMap.put(matchWord, freq - 1);
+                    foundMatch = true;
+                    break;
+                }
+            }
+            if (!foundMatch) {
+                return false;
+            }
+        }
+        return true;
+
+    }
+
+    public static void main(String[] args) {
```

Comment:
This is a good suite of tests. You could have a comment for each one to indicate what exactly it is testing instead of the wide blocks. And as you do most tests both ways, you could have a helper method that checks for the commutativity of the methods; given that (s1, s2) == (s2, s1)